crates/store/src/db/mod.rs

```diff
@@ -63,6 +63,7 @@ pub struct NoteRecord {
     pub note_id: RpoDigest,
     pub metadata: NoteMetadata,
     pub details: Option<Vec<u8>>,
+    pub nullifier: Option<Nullifier>,
```

Comment:
I'm not sure we actually need to have `nullifier` field here. The main reason why we want it in the table so that we could run a query that sets `consumed` for records which match a set of nullifiers. But other than that, we shouldn't really care about it (and the nullifier should be implicitly stored in the `details` field anyway).

Reply:
Right but how will we find out which notes to consume? We need some way of linking the nullifier to note ID.

Reply:
Do you mean how to identify network notes which haven't been consumed yet? Wouldn't this be something like:
```SQL
Select ... from notes
where
    execution_mode = 0 AND consumed = 0
```

Reply:
I mean when nullifiers are applied as part of a block, how do I mark their notes as being consumed. I need something where I can the equivalent of

```sql
UPDATE notes SET consumed = TRUE WHERE nullifier in rarray(?block.nullifiers)
```

Reply:
Yes, this looks correct. But why does this mean that `NoteRecord` needs a `nullifier` field? (we need the field in the table, but can probably skip it for the struct).

Reply:
Oh my bad; misunderstood.

Mostly the issue is that `NoteRecord` is used for both inserting and reading from the db, and it felt pretty difficult to separate. And for inserting we of course need the nullifier (unless we want to recalculate it for each one?).

Reply:
I've adapted it so the nullifier is passed along in addition to the `NoteRecord` for insertion. This feels a little disjointed to me; but I think we can leave it as is until we organise things a bit more in general.